### PR TITLE
Fix #59601: AWS: Check error code returned from describeVolume

### LINKED ISSUE #59601: Refactor checkIfAttachedToNode to return different error type based on error
Currently, this function just returns same `fmt.Errorf` no matter what error is. I think, it is a good idea to refactor this function to return different types of errors so as caller of this function can use the error to differentiate between errors and take action.

cc @tsmetana 
/sig aws
/sig storage

### PR DESCRIPTION
**Pull request by tsmetana**
The errors returned by the describeVolume call are not all equal:
if the error is of InvalidVolume.NotFound type it does not necessarily
mean the desired operation cannot be finished successfully.

Fixes #59601

```release-note
NONE
```

### CODE REVIEW

**Comment by tsmetana** [comment]
@jsafrane PTAL when you have time...

File: pkg/cloudprovider/providers/aws/aws.go [hunk 1]
```diff
@@ -1780,12 +1794,27 @@ func (d *awsDisk) waitForAttachmentStatus(status string) (*ec2.VolumeAttachment,
 	err := wait.ExponentialBackoff(backoff, func() (bool, error) {
 		info, err := d.describeVolume()
 		if err != nil {
+			// The VolumeNotFound error is special -- we don't need to wait for it to repeat
+			if isAWSErrorVolumeNotFound(err) {
+				if status == "detached" {
+					// The disk doesn't exist, assume it's detached, log warning and stop waiting
+					glog.Warningf("Waiting for volume %q to be detached but the volume does not exist", d.awsID)
+					stateStr := "detached"
+					attachment.State = &stateStr
```
Comment (jsafrane) [inline pkg/cloudprovider/providers/aws/aws.go]: `attachment` can be unitinialized here and `attachment.State` panics

**Pull request merged**